### PR TITLE
MM-19337 Enable users to view archived channels

### PR DESCRIPTION
**Pull request by valentijnnieman**
#### Summary
This adds an action for getting archived channels in the redux store (from the new endpoint), and modifies the searchChannel action to use the new search_archived endpoint if needed. 

- Has server changes: [PR 12995](https://github.com/mattermost/mattermost-server/pull/12995)

#### Ticket Link
[#12854 - MM-19337](https://github.com/mattermost/mattermost-server/issues/12854)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the [JavaScript driver](https://github.com/mattermost/mattermost-redux/blob/master/src/client/client4.js)

#### Test Information
This PR was tested on: Ubuntu 18

### CODE REVIEW

**Comment by valentijnnieman** [comment]
Not sure why my build got cancelled. I don't have permission to rerun the workflow unfortunately!

File: src/actions/channels.ts [hunk 1]
```diff
@@ -823,6 +823,37 @@ export function getChannels(teamId: string, page = 0, perPage: number = General.
     };
 }
 
+export function getArchivedChannels(teamId: string, page = 0, perPage: number = General.CHANNELS_CHUNK_SIZE): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        dispatch({type: ChannelTypes.GET_CHANNELS_REQUEST, data: null}, getState);
```
Comment (hmhealey) [inline src/actions/channels.ts]: Unless you're specifically looking at the state in `state.requests.channels.getChannels`, you can remove these lines that dispatch the `GET_CHANNELS_*` actions. We used to use them to track every type of request status, but we found that we don't use them very often

**Comment by valentijnnieman** [comment]
@hmhealey @saturninoabril I think this should be good now! Could you re-review when you have the time? Thanks :)

File: package-lock.json [hunk 2]
```diff
@@ -1560,7 +1560,7 @@
     },
     "array-equal": {
       "version": "1.0.0",
-      "resolved": "https://registry.npmjs.org/array-equal/-/array-equal-1.0.0.tgz",
+      "resolved": "http://registry.npmjs.org/array-equal/-/array-equal-1.0.0.tgz",
```
Comment (saturninoabril) [inline package-lock.json]: Please revert package-lock.json

File: src/client/client4.ts [hunk 3]
```diff
@@ -147,6 +147,10 @@ export default class Client4 {
         return `${this.getBaseRoute()}/channels`;
     }
 
+    getArchivedChannelsRoute() {
```
Comment (saturninoabril) [inline src/client/client4.ts]: I don't see this used elsewhere.

**Comment by valentijnnieman** [comment]
Thanks @saturninoabril for reviewing!

**Pull request merged**